runbotics/runbotics-desktop/src/action/desktop/desktop.action-handler.ts

```diff
@@ -0,0 +1,292 @@
+        }
+        return imagePath;
+    }
+
+    async findScreenRegion(input: DesktopFindScreenRegionActionInput): Promise<RegionObj> {
+        const image = input.imagePath;
+        this.checkImageExtension(image);
+
+        try {
+            const resource = await imageResource(image);
+            const region = await screen.find(resource); //optional param?
+            return this.toRegionObj(region);
+        } catch (error) {
+            throw new Error(error);
+        }
+    }
+
+    async waitForScreenRegion(input: DesktopWaitForScreenRegionActionInput): Promise<RegionObj> {
+        const image = input.imagePath;
+        this.checkImageExtension(image);
+
+        try {
+            const resource = await imageResource(image);
+            const region = await screen.waitFor(resource, 5000);
+            return this.toRegionObj(region);
+        } catch (error) {
+            throw new Error(error);
+        }
+    }
+
+    async readTextFromImage(input: DesktopReadTextFromImageActionInput): Promise<DesktopReadTextFromImageActionOutput> {
+        try {
+            const imageBuffer = fs.readFileSync(input.imagePath);
+
+            const worker = await createWorker();
+            await worker.loadLanguage(input.language);
+            await worker.initialize(input.language);
+    
+            const { data: { text } } = await worker.recognize(imageBuffer);
+            await worker.terminate();
+
+            return text;
+        } catch (error) {
+            throw new Error(error);
+        }
+    }
+
+    // async readCursorSelection(input: DesktopReadCursorSelectionActionInput): Promise<DesktopReadCursorSelectionActionOutput> {
+    //     this.checkPoint(input.startFirstCoordinate, input.startSecondCoordinate);
+    //     this.checkPoint(input.endFirstCoordinate, input.endSecondCoordinate);
+
+    //     await this.moveMouse(input.startFirstCoordinate, input.startSecondCoordinate);
+    //     await mouse.pressButton(Button.LEFT);
+    //     await this.moveMouse(input.endFirstCoordinate, input.endSecondCoordinate);
+    //     await mouse.releaseButton(Button.LEFT);
+    //     if (this.system === 'darwin') {
+    //         await keyboard.pressKey(Key.LeftCmd, Key.C);
+    //         await keyboard.releaseKey(Key.LeftCmd, Key.C);
+    //     } else {
+    //         await keyboard.pressKey(Key.LeftControl, Key.C);
+    //         await keyboard.releaseKey(Key.LeftControl, Key.C);
+    //     }
+    //     const content = await clipboard.getContent();
+    //     return content;
+    // }
+
+    async maximizeActiveWindow(): Promise<void> {
+        if (this.system === 'win32') {
+            // for Windows:
+            await this.performKeyboardShortcut([Key.LeftWin, Key.Up]);
+        } else if (this.system === 'darwin') {
+            // for macOS:
+            await this.performKeyboardShortcut([Key.LeftCmd, Key.Up]);
```

Comment:
I must sadden you, this keys combination does not work the same way on mac and linux too probably

Reply:
do you have any idea how this can universally work on macOS?

Reply:
imo the only way without any 3party app is `fn + f` shortcut